src/lib.rs

```diff
@@ -1,334 +1,201 @@
-        desc: &[protobuf::descriptor::FileDescriptorProto],
-        files_to_generate: &[String],
-        out_dir: &str,
-    ) {
-        use std::io::Write;
-
-        let output_dir = std::path::Path::new(out_dir);
-        let results = grpcio_compiler::codegen::gen(desc, &files_to_generate);
-        for res in results {
-            let out_file = output_dir.join(&res.name);
-            let mut f = std::fs::File::create(&out_file).unwrap();
-            f.write_all(&res.content).unwrap();
-        }
+    pub fn files<T: Into<String> + Clone>(&mut self, files: &[T]) -> &mut Self {
+        self.files = files
+            .iter()
+            .map(|t| t.clone().into())
+            .collect::<Vec<String>>();
+        self
     }
 
-    #[cfg(all(feature = "protobuf-codec", not(feature = "grpcio-protobuf-codec")))]
-    pub fn generate_grpcio(_: &[protobuf::descriptor::FileDescriptorProto], _: &[String], _: &str) {
+    pub fn includes(&mut self, includes: Vec<String>) -> &mut Self {
+        self.includes = includes;
+        self
     }
 
-    /// Convert protobuf files to use the old way of reading protobuf enums.
-    // FIXME: Remove this once stepancheg/rust-protobuf#233 is resolved.
-    pub fn replace_read_unknown_fields(out_dir: &str) {
-        let regex =
-            Regex::new(r"::protobuf::rt::read_proto3_enum_with_unknown_fields_into\(([^,]+), ([^,]+), &mut ([^,]+), [^\)]+\)\?").unwrap();
-        for f in fs::read_dir(out_dir).unwrap() {
-            let path = match f {
-                Ok(p) => p.path(),
-                Err(e) => panic!("failed to list {}: {:?}", out_dir, e),
-            };
-            if path.extension() != Some(std::ffi::OsStr::new("rs")) {
-                continue;
-            }
+    pub fn append_include(&mut self, include: String) -> &mut Self {
+        self.includes.push(include);
+        self
+    }
 
-            let mut text = String::new();
-            let mut f = File::open(&path).unwrap();
-            f.read_to_string(&mut text)
-                .expect("Couldn't read source file");
+    pub fn include_black_list(&mut self, include_black_list: Vec<String>) -> &mut Self {
+        self.include_black_list = include_black_list;
+        self
+    }
 
-            // FIXME Rustfmt bug in string literals
-            #[rustfmt::skip]
-            let text = {
-                regex.replace_all(
-                    &text,
-                    "if $1 == ::protobuf::wire_format::WireTypeVarint {\
-                        $3 = $2.read_enum()?;\
-                    } else {\
-                        return ::std::result::Result::Err(::protobuf::rt::unexpected_wire_type(wire_type));\
-                    }",
-                )
-            };
-            let mut out = File::create(&path).unwrap();
-            out.write_all(text.as_bytes())
-                .expect("Could not write source file");
-        }
+    pub fn append_black_listed_include(&mut self, include: String) -> &mut Self {
```

Comment:
How about just `black_list`?

Reply:
Hmm, I was worried that it was not clear what the black list was used for. I can use the shorter name and add some docs to explain.